build/cmake/FindUtf8Proc.cmake

```diff
@@ -0,0 +1,11 @@
+if(USE_BUNDLED_UTF8PROC)
+  message (STATUS "Enabling utf8proc (bundled)")
+  add_library (utf8proc_obj OBJECT "${UTF8PROC_SOURCES}")
+  set_property (TARGET utf8proc_obj PROPERTY POSITION_INDEPENDENT_CODE TRUE)
+  target_compile_definitions (utf8proc_obj PUBLIC UTF8PROC_STATIC)
+else ()
+  message (STATUS "Searching for utf8proc on system")
+  find_package(PkgConfig)
+  pkg_check_modules(PC_UTF8PROC REQUIRED libutf8proc IMPORTED_TARGET)
+  add_library(utf8proc ALIAS PkgConfig::PC_UTF8PROC)
```

Comment:
As written, this still links against the shared library (`PkgConfig::PC_UTF8PROC` -> `libutf8proc.so`) and propogates this requirement to downstream users. `find_library()` may still be required if we want to ensure linkage with the static library `libutf8proc.a` rather than the shared library `libutf8proc.so`. Support for static libraries via `IMPORTED_TARGET` [seems to be in a troublesome state](https://gitlab.kitware.com/cmake/cmake/-/issues/21714).

Note, this issue/complexity is predecated on the static library requirement. Confirming, is static linking actually an essential requirement for `utf8proc` library integration? If we do _not_ require static linking of `utf8proc`, the `IMPORTED_TARGET` solution should suffice as written.

Reply:
Since this codepath is most intended for use by distribution package maintainers, and that they will likely wish to use the dynamic library almost exclusively, I would consider that to be acceptable.